core/station/api/spec.did

```diff
@@ -592,6 +593,54 @@ type CreateExternalCanisterOperation = record {
   canister_id : opt principal;
 };
 
+type CanisterMethod = record {
+  // The canister to call.
+  canister_id : principal;
+  // The method to call on the canister.
+  method_name : text;
+};
+
+type CallExternalCanisterOperationInput = record {
+  // The canister method validating the argument blob:
+  //   - on validation success, returns a human-readable rendering of the argument blob
+  //     and then the request becomes `Created`;
+  //   - on validation error, returns a textual diagnostic message
+  //     and then the request becomes `Failed` with the textual diagnostic message as `reason`.
+  // Formally, the return type of the validation method must be
+  // ```
+  // variant {
+  //   Ok : text;
+  //   Err : text;
+  // }
+  // ```
+  // If omitted (`validation_method = null`), no validation of the argument blob is performed
+  // and no human-readable rendering of the argument blob is provided.
+  validation_method : opt CanisterMethod;
+  // The canister method that is called after the request becomes `Approved`
+  // passing the validated argument blob.
+  execution_method : CanisterMethod;
+  // The argument blob passed to both the validation and execution method.
+  arg : blob;
```

Comment:
should the argument be optional?

Reply:
Do you think it's gonna be common to call endpoints without any argument?

Reply:
only if you have triggering endpoints, e.g. `unlock`, `lock`, etc...

Reply:
Sure, I've made the argument blob optional now.